addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -2422,17 +2680,18 @@ var SnippetsMenu = Widget.extend({
      *
      * @private
      */
-    _onSnippetRemoved: function () {
-        this._disableUndroppableSnippets();
+    _onSnippetRemoved: async function (ev) {
+        await this._disableUndroppableSnippets();
         this._updateInvisibleDOM();
+        ev.data.onFinish();
     },
     /**
      * @private
      * @param {OdooEvent} ev
      */
     _onSnippetOptionVisibilityUpdate: async function (ev) {
         if (!ev.data.show) {
-            await this._activateSnippet(false);
+            this._enableLastEditor();
```

Comment:
Well it does make the snippet invisible. When you say it doesn't work anymore, I guess you mean the snippet options are still visible in the sidebar? I personally like that because then I can toggle again if I did it by mistake but I guess that's besides the point. This would be indeed a case of that `_enableLastEditor` vs `_deactivateSnippet` that I was mentioning above. I will look into it first thing Monday.

Reply:
You personally like the editor shows options for an element you can't see and preventing you to drop blocks after explicitly asking for the snippet to be hidden? ... ok but no sorry. Even if you wanted that, options may not work for an hidden snippet: what if an option has to computes its height or something like?

Reply:
I think it would make sense, if it worked properly. What you're telling me is that it's buggy if we let it. That's an argument to remove it, but not an argument to say it's a bad idea per se. I explicitly asked for the snippet to be _hidden_ indeed, not to be removed. As a user I just expect that to be a visual option for the snippet, not necessarily meaning that I can't edit it anymore. As for the option that requires to compute its height, nothing would technically prevent us to store that information in a state. This is a problem with relying on the DOM for everything imo.
In any case if it's the spec and doing otherwise breaks options to do that then we do that :shrug:

Reply:
@Zinston We had our Monday hijacked from us by al/fp so I guess you still have to look into this issue... first thing Thursday then ? :p

Reply:
Sure I'll look into it.